types/foundry/client/applications/apps/file-picker.d.mts

```diff
@@ -0,0 +1,329 @@
+    noResults: boolean;
+    selected: string;
+    source: { target: string; bucket?: string; buckets?: string[] };
+    sources: FilePicker["sources"];
+    target: string;
+    tileSize: number | null;
+    user: User;
+    favorites: Record<string, FavoriteFolder>;
+    buttons: FormFooterButton[];
+}
+
+/**
+ * The FilePicker application renders contents of the server-side public directory.
+ * This app allows for navigating and uploading files to the public path.
+ */
+export default class FilePicker extends HandlebarsApplicationMixin(ApplicationV2<FilePickerConfiguration>) {
+    static override DEFAULT_OPTIONS: DeepPartial<ApplicationConfiguration>;
+
+    static override PARTS: Record<string, HandlebarsTemplatePart>;
+
+    static override TABS: Record<string, ApplicationTabsConfiguration>;
+
+    /** The allowed values for the type of this FilePicker instance. */
+    static FILE_TYPES: ["any", "audio", "folder", "font", "graphics", "image", "imagevideo", "text", "video"];
+
+    /** Record the last-browsed directory path so that re-opening a different FilePicker instance uses the same target */
+    static LAST_BROWSED_DIRECTORY: string;
+
+    /** Record the last-configured tile size which can automatically be applied to new FilePicker instances */
+    static LAST_TILE_SIZE: number | null;
+
+    /** Record the last-configured display mode so that re-opening a different FilePicker instance uses the same mode. */
+    static LAST_DISPLAY_MODE: (typeof FilePicker.DISPLAY_MODES)[number];
+
+    /** Enumerate the allowed FilePicker display modes */
+    static DISPLAY_MODES: ["list", "thumbs", "tiles", "images"];
+
+    /** Cache the names of S3 buckets which can be used */
+    static S3_BUCKETS: string[] | null;
+
+    /** Return the upload URL to which the FilePicker should post uploaded files */
+    static get uploadURL(): string;
+
+    /** Retrieve the configured FilePicker implementation. */
+    static get implementation(): typeof FilePicker;
+
+    /**
+     *  @param [options={}] Options that configure the behavior of the FilePicker
+     */
+    constructor(options: ApplicationConfiguration & FilePickerConfiguration);
+
+    /** The full requested path given by the user */
+    request: string;
+
+    /**  A callback function to trigger once a file has been selected */
+    callback: Function | null;
+
+    /** The general file type which controls the set of extensions which will be accepted */
+    type: (typeof FilePicker.FILE_TYPES)[number];
+
+    /** The target HTML element this file picker is bound to */
+    field: HTMLElement | null;
+
+    /** A button controlling the display of the picker UI */
+    button: HTMLButtonElement | null;
+
+    /** The display mode of the FilePicker UI */
+    displayMode: (typeof FilePicker.DISPLAY_MODES)[number];
+
+    /** The file sources available for browsing */
+    sources: Record<Partial<FilePickerSource>, { target: string; bucket?: string; buckets?: string[] }>;
```

Comment:
?

Reply:
For reference: `game.data.files.storages` is `["data", "public"]` if s3 isn't configured. So the key could be omitted from the object.